Apps/Common/src/Services/PatientService.cs

```diff
@@ -94,34 +95,91 @@ public async Task<RequestResult<string>> GetPatientPhn(string hdid)
             return retVal;
         }
 
+        /// <inheritdoc/>
+        public async Task<string> GetPatientHdid(string phn)
+        {
+            using Activity? activity = Source.StartActivity();
+            RequestResult<PatientModel> patientResult = await this.GetPatient(phn, PatientIdentifierType.Phn).ConfigureAwait(true);
+            if (patientResult.ResultStatus != ResultType.Success || patientResult.ResourcePayload == null)
+            {
+                throw new ProblemDetailsException(
+                    new ProblemDetails
```

Comment:
You can use ExceptionUtility.CreateProblemDetails instead of doing new ProblemDetails. Pass in 'nameof<Your Class>' for typeName.

Reply:
fixed